### PR TITLE
[tweak] Internal logic for Reconstructed Alert Widget

### PR DESCRIPTION
**Pull request by hannahpurcell**
**Asana task**: [Internal logic for Reconstructed Alert Widget](https://app.asana.com/0/1185117109217413/1201879542889244)

Some of the work here reverts the changes made in the last PR because a less-invasive solution existed:
- instead of modifying `fetch_routes_at_stop` in `route.ex` to be generic, I reverted those edits, built a new function that returned full route objects, and then renamed the original function to be `fetch_simplified_routes_at_stop`

Description

- [ ] Needs version bump?

### CODE REVIEW

**Review by jzimbel-mbta — COMMENTED** [review]
Looks good, just some suggestions to reduce the amount of code by a little bit!

File: lib/screens/routes/route.ex [hunk 1]
```diff
@@ -64,6 +62,38 @@ defmodule Screens.Routes.Route do
     end
   end
 
+  @doc """
+  Fetches route type of routes that serve the given stop. `today` is used to determine whether
```
[the remark this answers is not in the document]
Comment (hannahpurcell) [reply lib/screens/routes/route.ex:66]: Whoops yup. Stale comment. Fetches full route
```suggestion
  Fetches routes that serve the given stop.
```

File: lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex [hunk 2]
```diff
@@ -15,21 +16,71 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ReconstructedAlert do
...
       ) do
-    with {:ok, routes_at_stop} <- fetch_routes_at_stop_fn.(stop_id, now, :subway),
+    with {:ok, routes_at_stop} <- fetch_routes_by_stop_fn.(stop_id, now, [:subway, :light_rail]),
          route_ids_at_stop = Enum.map(routes_at_stop, & &1.route_id),
-         {:ok, alerts} <- fetch_alerts_fn.(route_ids: route_ids_at_stop) do
+         {:ok, alerts} <- fetch_alerts_fn.(route_ids: route_ids_at_stop),
+         {:ok, stop_sequences} <- fetch_stop_sequences_by_stop_fn.(stop_id, route_ids_at_stop) do
+      station_sequences =
+        stop_sequences
+        |> Enum.map(fn stop_group ->
```
Comment (hannahpurcell) [inline lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex:32]:
```suggestion
        |> Enum.map(fn stop_sequence ->
```

File: lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex [hunk 3]
```diff
@@ -15,21 +16,71 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ReconstructedAlert do
...
+              :error -> nil
+            end
+          end)
+        end)
+        # Dedup the stop sequences (both directions are listed, but we only need 1)
+        |> Enum.reduce([], fn stop_group, acc ->
+          if Enum.find(acc, fn group -> Enum.sort(group) === Enum.sort(stop_group) end),
+            do: acc,
+            else: acc ++ [stop_group]
+        end)
```
[the remark this answers is not in the document]
Comment (hannahpurcell) [reply lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex:46]: 😮  that is _very good_

File: test/screens/routes/route_test.exs [hunk 4]
```diff
@@ -36,7 +36,73 @@ defmodule Screens.Routes.RouteTest do
...
+  end
+
+  describe "fetch_routes_by_stop/3" do
+    setup do
+      active_routes = [route_json("22"), route_json("44")]
+
+      all_routes = [route_json("22"), route_json("29"), route_json("44")]
+
+      stop_id = "1265"
+      today = ~D[2021-01-01]
```
Comment (hannahpurcell) [inline test/screens/routes/route_test.exs:91]:
```suggestion
      now = ~U[2021-01-01T00:00:00Z]
```

File: lib/screens/v2/widget_instance/alert.ex [hunk 5]
```diff
@@ -301,6 +314,15 @@ defmodule Screens.V2.WidgetInstance.Alert do
   def route_type(%__MODULE__{screen: %Screen{app_id: :bus_eink_v2}}), do: :bus
   def route_type(%__MODULE__{screen: %Screen{app_id: :gl_eink_v2}}), do: :light_rail
 
+  def route_type(%{
```
[the remark this answers is not in the document]
Comment (hannahpurcell) [reply lib/screens/v2/widget_instance/alert.ex:317]: Yeahhhhhhh.... Lot more misc code changes but it would make sense...

**Review by jzimbel-mbta — APPROVED** [review]
🥳

**Pull request merged**